apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp

```diff
@@ -68,6 +68,12 @@
     boolean isPasswordProvisionEnabled = Boolean.parseBoolean(request.getParameter("passwordProvisionEnabled"));
     boolean isSaaSApp = Boolean.parseBoolean(request.getParameter("isSaaSApp"));
     String callback = Encode.forHtmlAttribute(request.getParameter("callback"));
+    String emailUsernameEnable = application.getInitParameter("EnableEmailUserName");
+    if (StringUtils.isNotBlank(emailUsernameEnable) && Boolean.parseBoolean(emailUsernameEnable)) {
+        if (StringUtils.countMatches(username, "@") == 1) {
```

Comment:
Can we get "@" from a constant? This is already defined in the framework as well as the kernel.

https://github.com/wso2/carbon-identity-framework/blob/fa9d46e4371f0d4639b0f2174aae704f85cca19a/components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointConstants.java#L95

https://github.com/wso2/carbon-kernel/blob/da36ed2fc9bb539bd16f3b4209c7dcc101bb1737/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UserCoreConstants.java#L94

Reply:
Fixed.